client/src/cmdusart.c

```diff
@@ -270,39 +282,48 @@ static int CmdUsartBtFactory(const char *Cmd) {
     ret = usart_txrx((uint8_t *)string, strlen(string), data, &len, 1000);
     if (ret == PM3_SUCCESS) {
         PrintAndLogEx(SUCCESS, "RX (%3zu):%.*s", len, (int)len, data);
-        if (strcmp((char *)data, "OK+ROLE:S") == 0) {
+        if (strstr((char *)data, "OK")) {
             PrintAndLogEx(SUCCESS, "Role set to " _GREEN_("Slave"));
         } else {
             PrintAndLogEx(WARNING, "Unexpected response to AT+ROLE=S: " _YELLOW_("%.*s"), (int)len, data);
         }
     } else {
-        PrintAndLogEx(WARNING, "Lost contact with add-on, please try again");
+        PrintAndLogEx(WARNING, "Lost contact with add-on. Abort. \nmaybe AT command are different? Please check module's manual and report to us, or use 'usart txrx' manually.");
         return PM3_ESOFT;
     }
 
+    //ok give it another rest
+    msleep(500);
+
     memset(data, 0, sizeof(data));
     len = 0;
-    string = "AT+PIN1234";
+    string = "AT+PIN=1234";
     PrintAndLogEx(SUCCESS, "TX (%3zu):%.*s", strlen(string), (int)strlen(string), string);
 
     ret = usart_txrx((uint8_t *)string, strlen(string), data, &len, 1000);
     if (ret == PM3_SUCCESS) {
         PrintAndLogEx(SUCCESS, "RX (%3zu):%.*s", len, (int)len, data);
-        if (strcmp((char *)data, "OKsetPIN") == 0) {
+        if (strstr((char *)data, "OK")) {
             PrintAndLogEx(SUCCESS, "PIN set to " _GREEN_("1234"));
         } else {
             PrintAndLogEx(WARNING, "Unexpected response to AT+PIN: " _YELLOW_("%.*s"), (int)len, data);
         }
     } else {
-        PrintAndLogEx(WARNING, "Lost contact with add-on, please try again");
+        PrintAndLogEx(WARNING, "Lost contact with add-on. Abort. \nmaybe AT command are different? Please check module's manual and report to us, or use 'usart txrx' manually.");
         return PM3_ESOFT;
     }
 
     // parity must be changed before baudrate
+    
+    // hc-04 need AT+BAUD=115200,N\r\n
+    // 115200 means baud and N mean no parity
+
+    // hc-05 need AT+UART=115200,0,0,\r\n
+    // 115200 means baud, 0 mean 1 bit stop (?idk how to say it in english) and second 0 means no parity
     if (parity != USART_PARITY) {
         memset(data, 0, sizeof(data));
         len = 0;
-        string = "AT+PN";
+        string = "AT+P=N";
```

Comment:
I wonder, 
yours has a  `=`  in `AT+P=N`
and the old one doesn't.
`AT+PN*` 

That feels like a breaking change.